cmd/export/discover.go

```diff
@@ -201,18 +206,25 @@ func resourceToExtract(namespace string, labelSelector string, dynamicClient dyn
 	return resources, errors
 }
 
+func isAdmitted(clusterScopedRbac bool, resource metav1.APIResource) bool {
+	if !resource.Namespaced {
+		return clusterScopedRbac && slices.Contains(admittedClusterScopeKinds, resource.Kind)
+	}
+	return true
+}
+
 func getObjects(g *groupResource, namespace string, labelSelector string, d dynamic.Interface, logger logrus.FieldLogger) (*unstructured.UnstructuredList, error) {
 	c := d.Resource(schema.GroupVersionResource{
 		Group:    g.APIGroup,
 		Version:  g.APIVersion,
 		Resource: g.APIResource.Name,
 	})
-	if !g.APIResource.Namespaced {
-		return &unstructured.UnstructuredList{}, nil
-	}
-
 	p := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
-		return c.Namespace(namespace).List(context.Background(), opts)
+		if g.APIResource.Namespaced {
+			return c.Namespace(namespace).List(context.Background(), opts)
+		} else {
+			return c.List(context.Background(), opts)
```

Comment:
You mean adding ListOptions so that, for instance, when we collect the ServiceAccounts we also collect the meaningful ClusterRoleBinding and ClusterRole by using FieldSelector queries to fetch only the relevant instances?
If this is the purpose of the comment, it would require more changes to the discovery module because ATM if fetches only resources of a given, single kind. If we return resources of multiple kinds, instead, the code using getObjects function would need more changes.
Please let me know what was your actual goal.

Reply:
FYI, not all the fields are eligible to be used in the ListOptions.FieldSelector, e.g. the following command:
oc get clusterrolebindings.rbac.authorization.k8s.io --field-selector roleRef.name=AAA
would actually return an error because of that:
Error from server (BadRequest): Unable to find "rbac.authorization.k8s.io/v1, Resource=clusterrolebindings" that match label selector "", field selector "roleRef.name=AAA": "roleRef.name" is not a known field selector: only "metadata.name", "metadata.namespace"

Reply:
I think what I was asking, is that we look at the kind, and only retrieve the kinds of RBAC and SCC rather than filter here.